### ISSUE TITLE
Extra Information / Merging

### ISSUE THREAD

**Issue by cbizon** [NCATSTranslator/TranslatorArchitecture#77]
We have been discussing the question of extra information in TRAPI, which comes up in several contexts including creative mode and merging of results across ARAs.   Over the course of several weeks, we have developed three options:

Option 0b: A simple option in which inference provenance is not fully tracked ❤️ 
Option 1: Include all extra and merged node and edges in the normal node bindings along with extra provenance information 🚀 
Option 2: include extra information grouped together in "analysis" or "explanations" inside of a result 👀 
Option 3: I don't have enough information; I need more worked examples 👎 

These slides: https://docs.google.com/presentation/d/1CPJUEuWh-WxU-Fta3kof6drQC5FJX5MBcRnlDKnjog8/edit#slide=id.g146ea2e88f5_0_47 run through two examples and show what the TRAPI looks like in each case.

As pointed out on the call, the examples are written using dummy nodes, but could be updated to not make that assumption - the role of dummy nodes is not really what we're voting on here.  If one of the options with dummy nodes wins, that will be a subsequent discussion.

So please vote on one of the options above adding the emoji next to your favored option on this comment.   Feel free to advocate for your preferred solution below.

**Comment by edeutsch** [comment]
My opinion is this: there are only two serious options on the table here (1 and 2), and the “ARAX way”, where the QueryGraph is altered to reflect the topography of the answer is not visible to anyone voting (it is implicitly hidden in 1 but I don't think this is widely understood). I think it would be better to have a 3-way vote between the 3 main options that have been proposed. I also suggest that we tackle this issue at the Relay in an extended discussion; attempting to decide this in 40-minute increments is not very efficient or effective IMO.

**Comment by cbizon** [comment]
By 3 way vote do you mean splitting option 1 into one version with dummy indices and one with adding to the qgraph?   Would you be willing to update the slides to include the 'ARAX way'?

I don't mind restarting the vote once the slides better represent the approaches, but I honestly don't know what's left to say about these, I feel like we've discussed them to death and just need to pick.

**Comment by cbizon** [comment]
I think that the slide update would be adding NA and _extra1, _extra2, and _extra3 to the picture of the query graph on slides 6 and 10 is what the changes would be?  Is that correct?

**Comment by CaseyTa** [comment]
@edeutsch, Would the modified qgraph approach be able to support merged responses from multiple ARAs with different topographies?

**Comment by cbizon** [comment]
@CaseyTa yes it can - the qgraph would contain the union of all the different topographies across all ARAs and results.   In each result, it's possible that not every of the new qgraph nodes/edges would be bound in that particular result.

**Comment by CaseyTa** [comment]
@cbizon Thanks. I was initially thinking this would be messy for determining which are the distinct topographies (when looking at the qgraph alone), especially if any ARAs use branched logic in their query, but this is probably no more confusing than _not_ modifying the qgraph.

**Comment by cbizon** [comment]
Yes, and like I said above, the original idea is that the decision of modifying or not the qgraph is a second order question that can be decided on after the initial decision if option 1 wins.  But I think that @edeutsch and @dkoslicki disagree and would rather see an option that explicitly shows modification of the qgraph.

**Comment by mbrush** [comment]
Adding links to recent documents outlining proposals for representing extra information in TRAPI message, as discussed on Nov 17 and Dec 1 TRAPI calls. 
- LucidChart Diagrams illustrating a rich CM query and results merging scenario, on which data examples illustrating modeling proposals will be based - [see here](https://lucid.app/lucidchart/7f44b940-c9f9-4ade-bcd2-83b5e6182076/edit?viewport_loc=-1313%2C1785%2C2274%2C1155%2CGzS952H3pzZI&invitationId=inv_ecf1db03-cf9c-4e17-ad8a-22d78623961f) (you need to have/create Lucidchart account to view . . . ping me if you want a static pdf) 
- A data example representing this scenario as TRAPI json - [see here](https://github.com/NCATSTranslator/Evidence-Provenance-Confidence-Working-Group/blob/master/result_merging_examples/ara_result_merging_example_v2.json) 
     - . . . as of Dec 1 only the message sent form ARAGORN to the ARS in the scenario is provided.  The messages from ARAX to the ARS, and from the ARS to the UI (which holds the merged Result and supporting / extra info) are forthcoming.
